Bottom: 9fc678ad233ed078aa6dd70b1fdd08246a266644
Top:    17566a5ed8e0a7941077a8c3dc4259304f96c9bd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 20:18:13 +0000

Various HIDs: Add missing member pointers for the watch.. and block hander methods

These were missed when adding DBus support, since the initialisation of the
HID structure assumes that unspecified members are zero, and all subsequent
function pointers were explicitly 0, so as not to cause a type mismatch.



---

diff --git a/src/hid/bom/bom.c b/src/hid/bom/bom.c
index 19ad149..059e84e 100644
--- a/src/hid/bom/bom.c
+++ b/src/hid/bom/bom.c
@@ -570,6 +570,10 @@ HID bom_hid = {
   0,				/* bom_set_crosshair */
   0,				/* bom_add_timer */
   0,				/* bom_stop_timer */
+  0,				/* bom_watch_file */
+  0,				/* bom_unwatch_file */
+  0,				/* bom_add_block_hook */
+  0,				/* bom_stop_block_hook */
   0,				/* bom_log */
   0,				/* bom_logv */
   0,				/* bom_confirm_dialog */
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index babae3f..1557228 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -1103,6 +1103,10 @@ static HID gerber_hid = {
   gerber_set_crosshair,
   0 /* gerber_add_timer */ ,
   0 /* gerber_stop_timer */ ,
+  0 /* gerber_watch_file */ ,
+  0 /* gerber_unwatch_file */ ,
+  0 /* gerber_add_block_hook */ ,
+  0 /* gerber_stop_block_hook */ ,
   0 /* gerber_log */ ,
   0 /* gerber_logv */ ,
   0 /* gerber_confirm_dialog */ ,
diff --git a/src/hid/lpr/lpr.c b/src/hid/lpr/lpr.c
index f7f6c3e..5fcc9a7 100644
--- a/src/hid/lpr/lpr.c
+++ b/src/hid/lpr/lpr.c
@@ -147,6 +147,10 @@ HID lpr_hid = {
   0 /* lpr_set_crosshair */ ,
   0 /* lpr_add_timer */ ,
   0 /* lpr_stop_timer */ ,
+  0 /* lpr_watch_file */ ,
+  0 /* lpr_unwatch_file */ ,
+  0 /* lpr_add_block_hook */ ,
+  0 /* lpr_stop_block_hook */ ,
   0 /* lpr_log */ ,
   0 /* lpr_logv */ ,
   0 /* lpr_confirm_dialog */ ,
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index 2180325..964265a 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -1081,6 +1081,10 @@ HID             nelma_hid = {
 	nelma_set_crosshair,
 	0 /* nelma_add_timer */ ,
 	0 /* nelma_stop_timer */ ,
+	0 /* nelma_watch_file */ ,
+	0 /* nelma_unwatch_file */ ,
+	0 /* nelma_add_block_hook */ ,
+	0 /* nelma_stop_block_hook */ ,
 	0 /* nelma_log */ ,
 	0 /* nelma_logv */ ,
 	0 /* nelma_confirm_dialog */ ,
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index c0fcb9e..3c688bc 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -1359,6 +1359,10 @@ HID png_hid = {
   png_set_crosshair,
   0 /* png_add_timer */ ,
   0 /* png_stop_timer */ ,
+  0 /* png_watch_file */ ,
+  0 /* png_unwatch_file */ ,
+  0 /* png_add_block_hook */ ,
+  0 /* png_stop_block_hook */ ,
   0 /* png_log */ ,
   0 /* png_logv */ ,
   0 /* png_confirm_dialog */ ,
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index 4dbeedb..3545ddf 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -644,6 +644,10 @@ static HID eps_hid = {
   eps_set_crosshair,
   0 /* eps_add_timer */ ,
   0 /* eps_stop_timer */ ,
+  0 /* eps_watch_file */ ,
+  0 /* eps_unwatch_file */ ,
+  0 /* eps_add_block_hook */ ,
+  0 /* eps_stop_block_hook */ ,
   0 /* eps_log */ ,
   0 /* eps_logv */ ,
   0 /* eps_confirm_dialog */ ,
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index f5ac07e..bfb2e09 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1182,6 +1182,10 @@ HID ps_hid = {
   ps_set_crosshair,
   0 /* ps_add_timer */ ,
   0 /* ps_stop_timer */ ,
+  0 /* ps_watch_file */ ,
+  0 /* ps_unwatch_file */ ,
+  0 /* ps_add_block_hook */ ,
+  0 /* ps_stop_block_hook */ ,
   0 /* ps_log */ ,
   0 /* ps_logv */ ,
   0 /* ps_confirm_dialog */ ,
